var/spack/repos/builtin/packages/py-torchgeo/package.py

```diff
@@ -49,13 +51,14 @@ class PyTorchgeo(PythonPackage):
     depends_on("py-kornia@0.6.4:0.6.9", when="@0.3", type=("build", "run"))
     depends_on("py-kornia@0.5.11:0.6.9", when="@0.2", type=("build", "run"))
     depends_on("py-kornia@0.5.4:0.6.9", when="@0.1", type=("build", "run"))
+    depends_on("py-lightning@1.8:2", when="@0.4.1:", type=("build", "run"))
     depends_on("py-matplotlib@3.3:3", type=("build", "run"))
     depends_on("py-numpy@1.17.2:1", type=("build", "run"))
-    depends_on("py-omegaconf@2.1:2", type=("build", "run"))
+    depends_on("py-omegaconf@2.1:2", when="@:0.4.0", type=("build", "run"))
     depends_on("py-packaging@17:21", when="@0.3", type=("build", "run"))
     depends_on("pil@6.2:9+zlib+jpeg+tiff", type=("build", "run"))
     depends_on("py-pyproj@2.2:3", type=("build", "run"))
-    depends_on("py-pytorch-lightning@1.5.1:1+extra", when="@0.4:", type=("build", "run"))
+    depends_on("py-pytorch-lightning@1.5.1:1+extra", when="@0.4.0", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("py-pytorch-lightning@1.5.1:1", when="@0.4.0", type=("build", "run"))
```
This was only needed to add tensorboard, which isn't really a dep anyway. I think we can remove this.